### PR TITLE
[v2] Zero-dependency toolchain (swift-tools 6.0, Int64 amounts, hand-rolled parser)

### PR DESCRIPTION
**Pull request by pacu**
## Summary
- Rewrite Package.swift as a zero-dependency manifest: swift-tools-version 6.0, platform floors raised to macOS 13 / iOS 16, drop swift-parsing, swift-case-paths, BigDecimal, BigInt, swift-custom-dump and the os()-conditional dependency branching.
- Reimplement Amount on checked Int64 zatoshi fixed-point arithmetic instead of BigDecimal; all initializer signatures preserved except init(decimal:), which now takes Foundation.Decimal. v1 parsing leniency (e.g. "123.", ".5") is intentionally kept as-is.
- Hand-roll the ZIP-321 URI parser as an internal ZParser<Output> substring-combinator (literal/charset/prefix matching, backtracking Optionally, "rest of input"), replacing swift-parsing; a mechanical, behavior-preserving port verified against the full test suite and the shared conformance corpus (14-entry expected-failure map byte-identical to before).
- RecipientAddress.ValidatingClosure and ParserContext gain explicit Sendable conformance for Swift 6's default strict concurrency checking.
- Drop remaining swift-custom-dump usage (XCTAssertNoDifference/expectNoDifference -> XCTAssertEqual) from the last three files depending on it; the package no longer depends on swift-custom-dump at all.
- CHANGELOG updated to document the zero-dependency rewrite under Unreleased.

## Stack
Part of the v2.0.0 stacked-PR chain. Based on #79; reviewed diff contains only this branch's commits.

## Testing
Each commit lands with `swift test` fully green; the 14-entry conformance expected-failure map stays unchanged throughout.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01VwH46quaghs7U3VUFV2d3b

### CODE REVIEW

**Review by pacu — COMMENTED** [review]
Ack

File: Sources/ZcashPaymentURI/Amount.swift [hunk 1]
```diff
@@ -6,10 +6,15 @@
...
+
 /// An *non-negative* decimal ZEC amount represented as specified in ZIP-321.
 /// Amount can be from 1 zatoshi (0.00000001) to the `maxSupply` of 21M ZEC (`21_000_000`)
-public struct Amount: Equatable {
+///
+/// - Note: internally this is represented as a checked `Int64` count of zatoshi
+/// (1 ZEC == 100_000_000 zatoshi) fixed-point value rather than an arbitrary-precision
+/// decimal type. This is sufficient because `maxSupply` (`MAX_MONEY` = 2_100_000_000_000_000
+/// zatoshi) is comfortably representable by `Int64` (max ~9.2 * 10^18).
+public struct Amount: Equatable, Sendable {
```
Comment (pacu) [inline Sources/ZcashPaymentURI/Amount.swift:19]: Does Int make sense in the context of a ZIP-321 Payment URI? should this be an unsigned integer?

Comment (pacu) [reply Sources/ZcashPaymentURI/Amount.swift:19]: Agreed — addressed in 1fea153. `Amount` (and `Zatoshi` in #82, plus everything downstream) is now backed by `UInt64`, mirroring the reference implementation's `Zatoshis` newtype which wraps `u64`. A ZIP-321 amount is non-negative by grammar (`amountparam = 1*DIGIT [ "." 1*8DIGIT ]`), so the unsigned backing makes negative counts unrepresentable by construction; negative decimal *strings* are still rejected with `negativeAmount` by the explicit sign checks that run before conversion, so the error surface is unchanged. The change is propagated through the whole stack (parser, model, property generators, docs) with the suite green and the 100% coverage gate passing at the tip.

One cross-language note for your consideration: the Kotlin sibling keeps `Long` + construction-time validation, because a public Kotlin `ULong` API name-mangles functions beyond reach of plain-Java consumers of the `zip321-jvm` artifact. Semantics are identical (both enforce `0..MAX_MONEY`); if you'd rather Kotlin also move to `ULong` at the cost of Java-consumer ergonomics, say the word on the Kotlin stack.

File: Sources/ZcashPaymentURI/Amount.swift [hunk 2]
```diff
@@ -19,100 +26,181 @@ public struct Amount: Equatable {
...
+    }
+
+    /// Manually parses a plain decimal `String` into its exact zatoshi `UInt64` representation,
+    /// using only checked integer arithmetic (no floating point, no arbitrary-precision decimal type).
+    ///
+    /// Accepted grammar (intentionally lenient, matching v1 behavior):
+    /// `["-"] 1*DIGIT ["." *8DIGIT]` or `["-"] *DIGIT "." 1*8DIGIT`
+    /// i.e. at least one of the whole/fractional parts must be present, but either may be empty
+    /// (`"123."` and `".5"` are both accepted) — ZIP-321 grammar tightening is deferred.
+    static func parseZatoshi(from string: String) throws -> UInt64 {
```
Comment (pacu) [inline Sources/ZcashPaymentURI/Amount.swift]: This function should accept allowed amount denominations only.  Errors must be thrown eagerly and as early as possible. negative amounts are caught by the `-` sign but deferred for later on which shouldn't be. We already know that negative amounts are not part of the ZIP-321 spec so they should be rejected

File: Sources/ZcashPaymentURI/Amount.swift [hunk 3]
```diff
@@ -6,10 +6,17 @@
...
 /// Amount can be from 1 zatoshi (0.00000001) to the `maxSupply` of 21M ZEC (`21_000_000`)
-public struct Amount: Equatable {
+///
+/// - Note: internally this is represented as a checked `UInt64` count of zatoshi
+/// (1 ZEC == 100_000_000 zatoshi) fixed-point value rather than an arbitrary-precision
+/// decimal type, mirroring the reference implementation's `Zatoshis` (a `u64` newtype):
+/// a ZIP-321 amount is non-negative by grammar, so the unsigned type makes negative
+/// values unrepresentable, and `maxSupply` (`MAX_MONEY` = 2_100_000_000_000_000
+/// zatoshi) is comfortably representable by `UInt64` (max ~1.8 * 10^19).
+public struct Amount: Equatable, Sendable {
```
Comment (pacu) [inline Sources/ZcashPaymentURI/Amount.swift:19]: Add tests to make sure that negative amounts are properly rejected

File: Sources/ZcashPaymentURI/Amount.swift (hunk 2 above)
Comment (pacu) [reply Sources/ZcashPaymentURI/Amount.swift]: Addressed in bb293da: `parseZatoshi` now rejects a leading `-` **eagerly with `negativeAmount`** as its very first step — before any digit, bounds, or precision work — and a leading `+` (equally outside the grammar) fails with `invalidTextInput`. The previously-deferred negative checks are removed entirely (`grep isNegative` comes back empty). The v1 leniency for `"123."`/`".5"` shapes is still deliberately preserved at this point in the stack and tightened to the strict `1*DIGIT ["." 1*8DIGIT]` grammar in #86.

File: Sources/ZcashPaymentURI/Amount.swift (hunk 3 above)
Comment (pacu) [reply Sources/ZcashPaymentURI/Amount.swift:19]: Added in bb293da + 9799143: `testNegativeAmountsAreRejectedEagerlyOnEveryPath` asserts the exact `negativeAmount` error for `-1`, `-0`, `-0.5`, `-21000001`, `-0.123456789`, `-`, and `-.` on the string path, plus the Double and Decimal paths (`AmountError` is now `Equatable` so tests compare exact errors). Beyond negatives, `Amount.swift` is now at **100% coverage — 62/62 regions, 15/15 functions, 144/144 lines** (llvm-cov): malformed shapes (`1.2.3`, `.`, `..`, `1,5`, `1e5`), non-finite doubles, the zero constant, and the eager-rounding arm are all exercised, and two structurally-untestable fallback branches were replaced with digit-by-digit accumulation. SwiftLint warnings across Sources/ are also resolved in 9799143 (the four `TODO`s now reference #84, which resolves the transparent-prefix questions with the verified version bytes).
